### PR TITLE
Re-organize config generation and add some tests

### PR DESCRIPTION
**Pull request by aomarks**
There have been some bugs recently around config generation, and I found the previous code organization pretty confusing (after some time away from it).

This PR re-organizes the code and adds some new modules. You may want to look at each commit separately. There's pretty much zero new logic here, 99% moving code blocks around.

Now we can write way simpler unit tests that take `argv` as input and check the `Config` object, where the `Config` object specifies the behavior of tachometer. Included two such tests.

### CODE REVIEW

File: src/defaults.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import {BrowserName} from './browser';
+import {LocalUrl, Measurement, RemoteUrl} from './types';
+
+export const windowWidth = 1024;
+export const windowHeight = 768;
+export const root = '.';
+export const browserName: BrowserName = 'chrome';
+export const sampleSize = 50;
+export const timeout = 3;
+export const horizons = ['0%'];
```
Comment (rictic) [inline src/defaults.ts]: consider using `as const` here

File: src/util.ts [hunk 2]
```diff
@@ -22,3 +23,19 @@ export function isHttpUrl(str: string): boolean {
...
+    const stat = await fsExtra.stat(path);
+    if (stat.isDirectory()) {
+      return 'dir';
+    }
+    if (stat.isFile()) {
+      return 'file';
+    }
+  } catch (e) {
+    if (e.code !== 'ENOENT') {
+      throw e;
```
Comment (rictic) [inline src/util.ts]: Slightly clearer as:

```typescript
   if (e.code === 'ENOENT') {
     return undefined;
   }
   throw e;
```

Comment (aomarks) [reply src/util.ts]: Done.

File: src/defaults.ts (hunk 1 above)
Comment (aomarks) [reply src/defaults.ts]: Neat. It made me copy it in a few places.

**Pull request merged**